src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
+
+type Color = 'red' | 'green' | 'blue';
+type Snape = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  shape: Snape,
+  color: Color,
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Snape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sidesSum = this.a + this.b + this.c;
+    const maxSide = Math.max(this.a, this.b, this.c);
+    const sumTwoSmallerSides = sidesSum - maxSide;
+
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+      || maxSide >= sumTwoSmallerSides
+    ) {
```

Comment:
Split this to 2 different contitions(1 - check if all numbers are positive, 2 - `maxSide >= sumTwoSmallerSides`) with 2 different error messages